habitat/tasks/nav/nav_task.py

```diff
@@ -455,7 +461,8 @@ def update_metric(self, *args: Any, episode, action, **kwargs: Any):
         )
 
         if (
-            action == self._sim.index_stop_action
+            hasattr(self._task, "is_stop_called")
```

Comment:
EmbodiedTask should always have an `is_stop_called` field that gets set to `false` on reset and init.  Avoids this clunkiness.

Reply:
@erikwijmans what if your task doesn't have notion of stopping. When success or finished episode is just reach a target to some proximity.

Reply:
That's fine.  There is nothing that says it ever has to be set true, but the `hasattr(self._task, "is_stop_called")` check before everything is clunky.  One other suggestion:

Add
```
@property
def is_stop_called(self):
    return getattr(self, "_is_stop_called", False)
```

to the EmbodiedTask class.  The actions can then set `self._is_stop_called = True`.

Reply:
Btw, with the current method, isn't `_task .is_stop_called`, if it exists, always true?  I don't see where it gets set to false or deleted.

Reply:
The Environment class has step count limit after which it will stop the episode. `is_episode_active` is going to be redefined once it will be inheritance.